openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -398,16 +419,24 @@ def process_response(self, request, response):
         except:  # pylint: disable=bare-except
             pass
 
+        verify_all = VERIFY_USER_CHANGE_UNCONDITIONAL.is_enabled()
+
+        # One of the two places this can be called from (desired future location).
+        if verify_all:
+            user_id_in_session = self.get_user_id_from_session(request)
```

Comment:
should we still gate this on cookie marked for deletion? if it is i dont think we need to verify

Reply:
Good question. One the one hand, I don't think it would do any harm to include logout requests in this check, since I think we would handle that properly. On the other hand, this would go against the comment about this "restoring old behavior".

I like having the check be unconditional, since it's easier to reason about and the code is simplre. What if I just update the feature toggle comment to note this difference?

Reply:
That makes sense to me. In the worst case if we find it too noisy because of logouts we can always turn it off and rejigger